lib/secret_keys.rb

```diff
@@ -180,23 +189,35 @@ def load_secrets!(path_or_stream)
   # @param [String] data file data to parse
   # @return [Hash] data parsed to a hash
   def parse_data(data)
+    @format = :json
     JSON.parse(data)
   rescue JSON::JSONError
+    @format = :yaml
     YAML.safe_load(data)
   end
 
   # Recursively encrypt all values.
-  def encrypt_values(values)
+  def encrypt_values(values, original)
     if values.is_a?(Hash)
       encrypted_hash = {}
       values.keys.each do |key|
-        encrypted_hash[key.to_s] = encrypt_values(values[key])
+        original_value = original[key] if original.is_a?(Hash)
+        encrypted_hash[key.to_s] = encrypt_values(values[key], original_value)
```

Comment:
This should be figured out at that start, e.g. if someone added a symbol type key